Json.More/JsonSerializerOptionsExtensions.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Json.More;
+
+// TODO: .NET 5+ would have these methods marked with `RequiresUnreferencedCodeAttribute` to warn against tree trimming
+/// <summary>
+/// Provides extension functionality for <see cref="JsonSerializerOptions"/>.
+/// </summary>
+public static class JsonSerializerOptionsExtensions
+{
+	/// <summary>
+	/// Returns the converter for the specified type.
+	/// </summary>
+	/// <typeparam name="T">The <see cref="Type"/> to convert.</typeparam>
+	/// <param name="options">The <see cref="JsonSerializerOptions"/> being used.</param>
+	/// <returns>An implementation of <see cref="JsonConverter<>"/> as determined by the provided options</returns>
+	public static JsonConverter<T> GetConverter<T>(this JsonSerializerOptions options)
+	{
+		return (JsonConverter<T>)options.GetConverter(typeof(T));
+	}
+
+	/// <summary>
+	/// Read and convert the JSON to T.
+	/// </summary>
+	/// <remarks>
+	/// A converter may throw any Exception, but should throw <cref>JsonException</cref> when the JSON is invalid.
+	/// </remarks>
+	/// <typeparam name="T">The <see cref="Type"/> to convert.</typeparam>
+	/// <param name="options">The <see cref="JsonSerializerOptions"/> being used.</param>
+	/// <param name="reader">The <see cref="Utf8JsonReader"/> to read from.</param>
+	/// <returns>The value that was converted.</returns>
+	public static T? Read<T>(this JsonSerializerOptions options, ref Utf8JsonReader reader)
+	{
+		return options.GetConverter<T>().Read(ref reader, typeof(T), options);
+	}
+
+	/// <summary>
+	/// Read and convert the JSON to an arbitrary type.
+	/// </summary>
+	/// <param name="options">The <see cref="JsonSerializerOptions"/> being used.</param>
+	/// <param name="reader">The <see cref="Utf8JsonReader"/> to read from.</param>
+	/// <param name="arbitraryType">The <see cref="Type"/> to convert to.</param>
+	/// <returns>The value that was converted.</returns>
+	public static object? Read(this JsonSerializerOptions options, ref Utf8JsonReader reader, Type arbitraryType)
+	{
+		var converter = (IArbitraryDeserializer)Activator.CreateInstance(typeof(ArbitraryDeserializer<>).MakeGenericType(arbitraryType));
+		return converter.Read(ref reader, options);
+	}
+
+
+	private interface IArbitraryDeserializer
+	{
+		object? Read(ref Utf8JsonReader reader, JsonSerializerOptions options);
+	}
+
+	private class ArbitraryDeserializer<T> : IArbitraryDeserializer
+	{
+		public object? Read(ref Utf8JsonReader reader, JsonSerializerOptions options)
+		{
+			var converter = (JsonConverter<T>)options.GetConverter(typeof(T));
+
+			return converter.Read(ref reader, typeof(T), options);
+		}
+	}
```

Comment:
I agree that these can be cached.  Looking again, I think it makes sense to do it here instead of another PR.

I think something like this would do:

- Change `IArbitraryDeserializer` to abstract class `ArbitraryDeserializer`
- Add private static `ConcurrentDictionary<Type, JsonSchema>` as a cache to abstract base
- Add static `GetConverter()` to abstract base

Reply:
Couldn't add the `GetConverter()` to the abstract base since it doesn't have the generic at that level, but I did add the cache and use an abstract base class as suggested.

Reply:
So the point of making the base type abstract is to house the cache in it.  Then the `.GetConverter()` method is defined on that.  It encapsulates everything into the deserializer.  Maybe that's overengineering it.

I think what you've done is fine, and the abstract base class can go back to an interface.

Reply:
Ah, I misunderstood; got you now! Added one more commit for this - it does make it cleaner. Thanks for the advice!